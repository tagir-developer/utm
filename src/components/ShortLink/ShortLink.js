import React from 'react'

const ShortLink = ({ text, title, shortLink, shortLinkChange, copyResultLink }) => {

	const id = 'short-link-input'

	const shortLinkHandler = () => {
		return copyResultLink(id)
	}

	return (
		<div className="mt-3">
			<label htmlFor="short-link-input" className="font-weight-bold">{title}</label>
			<div className="input-group">
				<div className="input-group-prepend">
					<button
						className="btn btn-primary"
						type="button"
						id="short-link-button"
						onClick={shortLinkHandler}
					>
						Скопировать
					</button>
				</div>
				<input
					id="short-link-input"
					type="text"
					className="form-control"
					placeholder={text}
					aria-label="Example text with button addon"
					aria-describedby="short-link-button"
					value={shortLink}
					onChange={e => shortLinkChange(e)}
				/>
			</div>
		</div>
	)
}

ShortLink.defaultProps = {
	text: 'Значение плейсхолдера', 
	title: 'Дефолтный заголовок', 
	shortLink: 'Короткая ссылка', 
	shortLinkChange: () => {}, 
	copyResultLink: () => {},
}

export default ShortLink