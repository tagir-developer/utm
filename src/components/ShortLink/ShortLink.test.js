import React from 'react'
import ShortLink from './ShortLink'

describe('<ShortLink />', () => {

	let wrapper

	beforeEach(() => {
		wrapper = shallow(<ShortLink />)
	})

	describe('Component render. Component props testing', () => {

		it('Should ShortLink component render with default props', () => {
			expect(toJson(wrapper)).toMatchSnapshot()
		})

		it('Should ShortLink component render with test props', () => {
			wrapper.setProps({
				text: 'Тестовый плейсхолдер',
				title: 'Тестовый заголовок',
				shortLink: 'Тестовая ссылка',
				shortLinkChange: () => {},
				copyShortLink: () => {},
			})
			expect(toJson(wrapper)).toMatchSnapshot()
		})

	})

	describe('ShortLink component handlers testing', () => {

		it('Should call "ShortLinkChange"', () => {
			const changeHandler = jest.fn()
			wrapper.setProps({
				shortLinkChange: changeHandler,
			})
			expect(changeHandler).toHaveBeenCalledTimes(0)
			wrapper.find('input').simulate('change')
			expect(changeHandler).toHaveBeenCalled()
		})

		it('Should call "copyResultLink"', () => {
			const changeHandler = jest.fn()
			wrapper.setProps({
				copyResultLink: changeHandler,
			})
			expect(changeHandler).toHaveBeenCalledTimes(0)
			wrapper.find('button').simulate('click')
			expect(changeHandler).toHaveBeenCalled()
		})

		it('Should use default handler "ShortLinkChange"', () => {
			const result = ShortLink.defaultProps.shortLinkChange()
			expect(result).toBe(undefined)
		})

		it('Should use default handler "copyResultLink"', () => {
			const result = ShortLink.defaultProps.copyResultLink()
			expect(result).toBe(undefined)
		})

	})


})