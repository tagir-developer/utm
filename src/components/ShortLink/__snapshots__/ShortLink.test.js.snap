// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShortLink /> Component render. Component props testing Should ShortLink component render with default props 1`] = `
<div
  className="mt-3"
>
  <label
    className="font-weight-bold"
    htmlFor="short-link-input"
  >
    Дефолтный заголовок
  </label>
  <div
    className="input-group"
  >
    <div
      className="input-group-prepend"
    >
      <button
        className="btn btn-primary"
        id="short-link-button"
        onClick={[Function]}
        type="button"
      >
        Скопировать
      </button>
    </div>
    <input
      aria-describedby="short-link-button"
      aria-label="Example text with button addon"
      className="form-control"
      id="short-link-input"
      onChange={[Function]}
      placeholder="Значение плейсхолдера"
      type="text"
      value="Короткая ссылка"
    />
  </div>
</div>
`;

exports[`<ShortLink /> Component render. Component props testing Should ShortLink component render with test props 1`] = `
<div
  className="mt-3"
>
  <label
    className="font-weight-bold"
    htmlFor="short-link-input"
  >
    Тестовый заголовок
  </label>
  <div
    className="input-group"
  >
    <div
      className="input-group-prepend"
    >
      <button
        className="btn btn-primary"
        id="short-link-button"
        onClick={[Function]}
        type="button"
      >
        Скопировать
      </button>
    </div>
    <input
      aria-describedby="short-link-button"
      aria-label="Example text with button addon"
      className="form-control"
      id="short-link-input"
      onChange={[Function]}
      placeholder="Тестовый плейсхолдер"
      type="text"
      value="Тестовая ссылка"
    />
  </div>
</div>
`;
